File: articles/calculus-for-systems-modeling/articles/functions-of-several-variables/r/multivariable_function_grid.R
system_response <- function(x, y) 3.0 * x + 2.0 * y + 0.5 * x * y
is_feasible <- function(x, y, budget = 10) x >= 0 & y >= 0 & x + y <= budget

grid <- expand.grid(x = seq(0, 10, by = 2), y = seq(0, 10, by = 2))
grid$output <- system_response(grid$x, grid$y)
grid$feasible <- is_feasible(grid$x, grid$y)
grid$warning <- ifelse(grid$feasible, "", "Input combination is outside the feasible region.")
dir.create("outputs/tables", recursive = TRUE, showWarnings = FALSE)
write.csv(grid, "outputs/tables/r_multivariable_function_grid.csv", row.names = FALSE)
print(grid)

File: articles/calculus-for-systems-modeling/articles/functions-of-several-variables/julia/multivariable_function_grid.jl
system_response(x, y) = 3.0*x + 2.0*y + 0.5*x*y
is_feasible(x, y; budget=10.0) = x >= 0 && y >= 0 && x + y <= budget
println("x,y,output,feasible,warning")
for x in 0:2:10, y in 0:2:10
    feasible = is_feasible(x, y)
    warning = feasible ? "" : "Input combination is outside the feasible region."
    println("$x,$y,$(system_response(x,y)),$feasible,$warning")
end
